R/credentials_gce.R

```diff
@@ -27,13 +27,22 @@ credentials_gce <- function(scopes = NULL, service_account = "default", ...) {
   if (!all(scopes %in% instance_scopes)) {
     return(NULL)
   }
-  credentials <- fetch_access_token(scopes, service_account = service_account)
+
+  # The access_token can only include the token itself, *not the expiration and type*. Otherwise, the
+  # httr code will create extra header lines that bust the POST request:
+  full_credentials <- fetch_access_token(scopes, service_account = service_account)
+  credentials = full_credentials$access_token
+
   params <- list(
     as_header = TRUE,
     scope = scopes,
     service_account = service_account
   )
-  token <- GceToken$new(credentials = credentials, params = params)
+  # The underlying Token2 class appears to *require* an endpoint and an app, though it doesn't
+  # use them for anything in this case...
```

Comment:
This is pretty believable to me, as I've found another instance of similar:

https://github.com/r-lib/httr/issues/576

Reply:
Looks good to me. Thanks!